api_modules/sessionize.py

```diff
@@ -1,5 +1,122 @@
 '''
-    Put code for Sessionize API acraping here. 
-    Update this docstring with details of how to use the module.
+    Sessionize API acraping here. 
+
+    created 28.01.2024 by bogdansharpy@gmail.com
+
+    Needs bs4 to be installed: pip install bs4
+
+    Uses publicly accessible Sessionize Embedding API : https://sessionize.com/playbook/embedding
+    API Endpoint:
+        f"https://sessionize.com/api/v2/{event_id}/view/{type}?under=True"
+        type in {"GridSmart", "SpeakerWall", "Speakers", "Sessions"}
+
+    Example of how to use script:
+        python sessionize.py --id amtv2kwb --name "PyCon 2022"
+
+    With no arguments by default will get this events: PyCon 2022, PyCon Limerick 2023, PyCon 2023
+
 '''
 
+import argparse
+import requests
+import csv
+from bs4 import BeautifulSoup
+
+class Sessionize:
+    def __init__(self, events=None) -> None:
+        default_events = { \
+            'amtv2kwb': 'PyCon 2022', \
+            'jb4vxosa': 'PyCon Limerick 2023', \
+            'jbshwhme': 'PyCon 2023', \
+        }
+        self.events = events if events else default_events
+
+    def get_sessionize_data(self):
+        for event_id, event_name in self.events.items():
+            for type in ("Sessions", "Speakers"):
+                self.__get_sessionize_event(event_id, event_name, type)
+
+    def __get_sessionize_event(self, event_id, event_name, type):
+        url = f"https://sessionize.com/api/v2/{event_id}/view/{type}?under=True"
+        csv_file_name = f"{event_name}_{type}.csv"
+        text_or_default = lambda el, default: el.text if el else default
+        rows = []
+
+        try:
+            response = requests.get(url)
+            response.raise_for_status()
+            soup = BeautifulSoup(response.content, 'html.parser')
+
+            if type == "Speakers":
+                rows.append(['id', 'name', 'tagline', 'bio', 'photo']) 
+                top_element = soup.find('ul', class_="sz-speakers--list")
+                if not top_element:
+                    raise RuntimeError('Unexpected server response!')
+                for row in top_element.find_all('li', class_='sz-speaker'):
+                    id = row.get('data-speakerid', '')
+                    name = text_or_default(row.find('h3', class_='sz-speaker__name'), "")
+                    tagline = text_or_default(row.find('h4', class_='sz-speaker__tagline'), "")
+                    bio = text_or_default(row.find('p', class_='sz-speaker__bio'), "")
+                    bio = bio.replace('<br>', '\n')
+                    photo = ''
+                    photo_el = row.find('div', class_='sz-speaker__photo')
+                    if photo_el:
+                        photo_img_el = photo_el.find('img')
+                        if photo_img_el:
+                            photo = photo_img_el.get('src', '')
+                    rows.append([id, name, tagline, bio, photo]) 
```

Comment:
Some of the string fields here appear with leading and/or trailing spaces in the exported CSV. Probably good to call `strip()` on these strings here to remove them.

Reply:
You are right. I am going to add strip()